### ISSUE TITLE
MemoryHighAction - cgroup systemd actions

### ISSUE THREAD

**Issue by dariodsa** [systemd/systemd#26212]
### Component

systemctl

### Is your feature request related to a problem? Please describe

In discussion with ChatGPT he suggested me that there is something like this:
```
[Service]
MemoryHigh=500M
MemoryHighAction=/usr/bin/some_script.sh
```
But as I searched over systemd code base and documentation I found nothing like that. That feature would be a great addition in the universe of systemd. So to sum up, I would like to have an action when systemd service crosses over `MemoryHigh` bar.

### The systemd version you checked that didn't have the feature you are asking for

252

**Comment by poettering** [comment]
I am not convinced by this. Invoking a process when we are under pressure is generally the opposite of what we should do under pressure. We should release resources, not run even more stuff.

I have a branch somewhere that adds a concept to the whole of systemd via the kernel PSI interface to release their caches (including glibc allocation caches) under pressure, but this works differently: we'd not spawn anything new, but hand the notification directly to the services themselves so that they can react to it without need to allocate any additional resources.

ultimately, flushing caches is something the apps have to do on their own inside their daemon code, hence invoking something external would usually mean that that extenral process would just notify the main daemon process in some way, and that raises the question: why bump this off through an external process at all.

**Comment by dariodsa** [comment]
@poettering, I agree with you on that. If application has trouble with memory, no external process should be called in order to help her. But, I was looking that from another perspective and that was from monitoring stand point. It is better for an external process to report "There is an issue with memory usage with service S1", than to have an process which will look every N seconds the memory usage of S1. That is already possible using ``eventfd`` with cgroups notifications so I thought that Systemd could encapsulate that.
